api/app/models/concerns/spree/webhooks/has_webhooks.rb

```diff
@@ -9,20 +9,34 @@ module HasWebhooks
         after_update_commit(proc { queue_webhooks_requests!(event_name(:update)) })
 
         def queue_webhooks_requests!(event)
-          return if disable_spree_webhooks? || updating_only_timestamps? || body.blank?
+          return if disable_spree_webhooks? || updating_only_timestamps?
+          return if (event_body = webhooks_body_for(event: event)).blank?
 
-          Spree::Webhooks::Subscribers::QueueRequests.call(event: event, body: body)
+          Spree::Webhooks::Subscribers::QueueRequests.call(event: event, body: event_body)
         end
       end
 
       private
 
-      def event_name(operation)
-        "#{self.class.name.demodulize.tableize.singularize}.#{operation}"
+      def webhooks_body_for(event:)
+        created_event = event_record(event)
+        created_event.reload
```

Comment:
why is this line necessary?

Reply:
I think that case is only necessary for the test suite @aleksandarpetrushev. When the event is created in line 22, its timestamps don't coincide with the ones in the database when checking it with AR:

```ruby
Spree::Webhooks::Event.where(name: event)
=> [#<Spree::Webhooks::Event:0x000055a95c90dac0
  id: 19,
  execution_time: nil,
  name: "variant.discontinued",
  request_errors: nil,
  response_code: nil,
  subscriber_id: nil,
  success: nil,
  url: nil,
  created_at: Fri, 19 Nov 2021 15:23:08.683798000 UTC +00:00,
  updated_at: Fri, 19 Nov 2021 15:23:08.683798000 UTC +00:00,
  preferences: nil>]
```

while the `created_event` variable holds this:

```ruby
created_event
=> #<Spree::Webhooks::Event:0x000055a95c8779d0
 id: 19,
 execution_time: nil,
 name: "variant.discontinued",
 request_errors: nil,
 response_code: nil,
 subscriber_id: nil,
 success: nil,
 url: nil,
 created_at: Fri, 19 Nov 2021 15:23:08.683798067 UTC +00:00,
 updated_at: Fri, 19 Nov 2021 15:23:08.683798067 UTC +00:00,
 preferences: {}>
```

as you can see, nor the `created_at` nor the `updated_at` match, but the id does, meaning they should refer to the same record. Just after a `reload` you get that:

```ruby
created_event.reload
=> #<Spree::Webhooks::Event:0x000055a95c8779d0
 id: 19,
 execution_time: nil,
 name: "variant.discontinued",
 request_errors: nil,
 response_code: nil,
 subscriber_id: nil,
 success: nil,
 url: nil,
 created_at: Fri, 19 Nov 2021 15:23:08.683798000 UTC +00:00,
 updated_at: Fri, 19 Nov 2021 15:23:08.683798000 UTC +00:00,
 preferences: nil>
```

Reply:
okay 👍

Reply:
By the way, this happens only with Postgres.